test_collections/sdk_tests/support/python_testing/models/test_case.py

```diff
@@ -0,0 +1,281 @@
+        self.step_unknown_count = 0
+        self.__runned = 0
+
+    def start(self, count: int) -> None:
+        pass
+
+    def stop(self, duration: int) -> None:
+        pass
+
+    def test_start(self, filename: str, name: str, count: int) -> None:
+        pass
+        # Dont know if it is necessary for python testing (came from chip_tool)
+        # self.next_step()
+
+    def test_stop(self, exception: Exception, duration: int) -> None:
+        self.current_test_step.mark_as_completed()
+
+    def step_skipped(self, name: str, expression: str) -> None:
+        self.current_test_step.mark_as_not_applicable(
+            f"Test step skipped: {name}. {expression} == False"
+        )
+        self.next_step()
+
+    def step_start(self, name: str) -> None:
+        pass
+
+    def step_success(self, logger: Any, logs: str, duration: int, request: Any) -> None:
+        self.__handle_logs(logs)
+        self.next_step()
+
+    def step_failure(
+        self, logger: Any, logs: str, duration: int, request: Any, received: Any
+    ) -> None:
+        self.__handle_logs(logs)
+        self.__report_failures(logger, request, received)
+        self.next_step()
+
+    def step_unknown(self) -> None:
+        self.__runned += 1
+
+    def is_finished(self) -> bool:
+        return self.test_finished
+
+    def __handle_logs(self, logs: Any) -> None:
+        for log_entry in logs or []:
+            if not isinstance(log_entry, MatterLog):
+                continue
+
+            test_engine_logger.log(
+                CHIPTOOL_LEVEL,
+                CHIP_LOG_FORMAT.format(log_entry.module, log_entry.message),
+            )
+
+    def __report_failures(self, logger: Any, request: TestStep, received: Any) -> None:
+        """
+        The logger from runner contains all logs entries for the test step, this method
+        seeks for the error entries.
+        """
+        if not logger:
+            # It is expected the runner to return a PostProcessResponseResult,
+            # but in case of returning a different type
+            self.current_test_step.append_failure(
+                "Test Step Failure: \n Expected: '<Empty>' \n Received:  '<Empty>'"
+            )
+            return
+
+        # Iterate through the entries seeking for the errors entries
+        for log_entry in logger.entries or []:
+            if log_entry.is_error():
+                # Check if the step error came from exception or not, since the message
+                # in exception object has more details
+                # TODO: There is an issue raised in SDK runner in order to improve the
+                # message from log_entry:
+                # https://github.com/project-chip/connectedhomeip/issues/28101
+                if log_entry.exception:
+                    self.current_test_step.append_failure(log_entry.exception.message)
+                else:
+                    self.current_test_step.append_failure(log_entry.message)
```

Comment:
Is this relevant for python tests? Or does it only make sense for chip-tool/app test cases that use the websocket runner? Do the python tests implement this `logger`?